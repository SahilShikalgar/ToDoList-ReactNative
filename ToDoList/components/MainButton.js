import React from 'react'
import { TouchableOpacity, TouchableNativeFeedback, Platform, StyleSheet, View, Text } from 'react-native';
import Colors from '../constants/Colors';

export default function MainButton(props) {
    let ButtonComponent = TouchableOpacity;

    if (Platform.Version >= 21) {
        ButtonComponent = TouchableNativeFeedback;
    }

    return (
        <View style={styles.buttonContainer}>
            <ButtonComponent activeOpacity={0.6} onPress={props.onPress}>
                <View style={styles.button}>
                    <Text style={styles.buttonText}>{props.children}</Text>
                </View>
            </ButtonComponent>
        </View>
    );
}

const styles = StyleSheet.create({
    buttonContainer: {
        borderRadius: 25
    },
    button: {
        backgroundColor: Colors.primary,
        paddingVertical: 15,
        paddingHorizontal: 45,
        borderRadius: 25
    },
    buttonText: {
        color: 'white',
        fontFamily: 'lato',
        fontSize: 24,
        textAlign: 'center'
    }
});
