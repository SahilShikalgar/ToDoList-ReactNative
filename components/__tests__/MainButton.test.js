import React from 'react';
import { shallow } from 'enzyme';
import MainButton from '../MainButton';
import { TouchableOpacity, TouchableNativeFeedback } from 'react';

describe('Main Button Component', () => {
    const goToMainScreen = jest.fn();
    const Ionicons = jest.fn();
    const component = shallow(
        <MainButton onPress={goToMainScreen} Ionicons={Ionicons}>
            NEXT
        </MainButton>
    );

    it('should renderer properly', () => {
        expect(component).toMatchSnapshot();
    });

    it('should render TouchableNativeFeedback for Platform.Version >= 21', () => {
        // console.log(component);
        // expect(typeof component.ButtonComponent).toBe(TouchableOpacity);
        console.log(component);
    });
});
