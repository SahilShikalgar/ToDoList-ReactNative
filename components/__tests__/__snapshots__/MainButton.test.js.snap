// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Button Component should renderer properly 1`] = `ShallowWrapper {}`;
